Move DD4Hep workflows from 13TeV to 14TeV

silviodonato opened the pull request:
This PR moves TTbar and ZMM DD4Hep workflows from 13 TeV to 14 TeV, as discussed today at the ORP meeting. The reason is that `11634.0` is used in the current RelVal production and then we want the corresponding `11634.911`. Same thing for `11642.911` which has been moved to 14 TeV `11650.911`. The short matrix has been updated accordingly.
The PR has been tested simply with `runTheMatrix.py -l 11634.911,11650.911`

silviodonato committed:
add TTbar_14TeV DD4HEP workflow

silviodonato commented:
please test 11634.911

cc: @cms-sw/simulation-l2 @cms-sw/geometry-l2

davidlange6 commented on Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:
```diff
@@ -950,7 +950,7 @@ def setup_(self, step, stepName, stepDict, k, properties):
             stepDict[stepName][k] = merge([{'--geometry' : dd4hepGeom, '--procModifiers': 'dd4hep'}, stepDict[step][k]])
     def condition(self, fragment, stepList, key, hasHarvest):
-        return ((fragment=='TTbar_13' or fragment=='ZMM_13' or fragment=='SingleMuPt10') and '2021' in key) \
+        return ((fragment=='TTbar_13' or fragment=='TTbar_14' or fragment=='ZMM_13' or fragment=='SingleMuPt10') and '2021' in key) \
```
since we won't use dd4hep for run2, perhaps 13 tev workflows aren't very useful?

silviodonato replied:
Since we have only 11624.911 in pre2, I think it is good to have both 11624.911 and 11634.911 in pre3

chayanit replied:
I also agreed with @davidlange6 that 13TeV isn't useful to keep

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32857/21079

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #32765, #31854, #27983, #32128
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #32765, #31854, #27983, #32128, #32843

cmsbuild commented:
A new Pull Request was created by  @silviodonato (Silvio Donato) for master.

It involves the following packages:

Configuration/PyReleaseValidation


@jordan-martins, @chayanit, @wajidalikhan, @kpedro88, @cmsbuild, @srimanob can you please review it and eventually sign? Thanks.
@makortel, @Martin-Grunewald, @fabiocos, @slomeo this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

cvuosalo commented:
@cmsbuild please test

cvuosalo commented:
@silviodonato Thank you for creating this workflow.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-1d68c5/12808/summary.html
**COMMIT**: 01b39f0d596f128a94ca430b4c283a6b118e5899
**CMSSW**: CMSSW_11_3_X_2021-02-09-1100/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32857/12808/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 37
 - DQMHistoTests: Total histograms compared: 2751765
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2751742
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 36 files compared)
 - Checked 156 log files, 37 edm output root files, 37 DQM output files

chayanit commented on Configuration/PyReleaseValidation/python/relval_2017.py:
```diff
@@ -55,7 +55,7 @@
            10824.511,
            10824.521,
-           11624.911, 11642.911,
+           11634.911,11624.911, 11642.911,
```
@silviodonato , @cvuosalo could you clarify why we need 13TeV for Run3 scenario? Shall we move everything about DD4HEP to 14TeV (also ZMM)?

davidlange6 commented:
For what purpose? Presumably the goal is to compare a dd4hep geometry to a dd geometry - the latter has 14 TeV samples for run3.

> On Feb 9, 2021, at 7:03 PM, Silvio Donato <notifications@github.com> wrote:
> 
> 
> @silviodonato commented on this pull request.
> 
> In Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py:
> 
> > @@ -950,7 +950,7 @@ def setup_(self, step, stepName, stepDict, k, properties):
>              dd4hepGeom+=stepDict[step][k]['--geometry']
>              stepDict[stepName][k] = merge([{'--geometry' : dd4hepGeom, '--procModifiers': 'dd4hep'}, stepDict[step][k]])
>      def condition(self, fragment, stepList, key, hasHarvest):
> -        return ((fragment=='TTbar_13' or fragment=='ZMM_13' or fragment=='SingleMuPt10') and '2021' in key) \
> +        return ((fragment=='TTbar_13' or fragment=='TTbar_14' or fragment=='ZMM_13' or fragment=='SingleMuPt10') and '2021' in key) \
> 
> Since we have only 11624.911 in pre2, I think it is good to have both 11624.911 and 11634.911 in pre3
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or unsubscribe.
>

silviodonato committed:
remove 11624.911, add 11634.911 in the short matrix, move ZMM DD4HEP to 14TeV

silviodonato commented:
please test workflows 11634.911, 11650.911

silviodonato commented:
As you prefer, I've removed the 13TeV workflow, and so I've also updated the short matrix to replace the DD4HEP worflow from `11624.911` to `11634.911`. I also updated the ZMM to 14TeV, just to have everything at 14TeV (`11642.911` -> `11650.911`)

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-32857/21093

- This PR adds an extra 24KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  Configuration/PyReleaseValidation/python/relval_2017.py  modified in PR(s): #32765, #31854, #27983, #32128
  - File  Configuration/PyReleaseValidation/python/upgradeWorkflowComponents.py  modified in PR(s): #32765, #31854, #32843, #27983, #32128

cmsbuild commented:
Pull request #32857 was updated. @chayanit, @srimanob, @wajidalikhan, @kpedro88, @jordan-martins can you please check and sign again.

cmsbuild commented:
+1

**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-1d68c5/12826/summary.html
**COMMIT**: 53711db319bca172673ff2996d953e7adb36d11d
**CMSSW**: CMSSW_11_3_X_2021-02-09-2300/slc7_amd64_gcc900
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/32857/12826/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Comparison Summary



@slava77 comparisons for the following workflows were not done due to missing matrix map:
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-1d68c5/11634.911_TTbar_14TeV+2021_DD4hep+TTbar_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA
   * /data/cmsbld/jenkins/workspace/compare-root-files-short-matrix/data/PR-1d68c5/11650.911_ZMM_14+2021_DD4hep+ZMM_14TeV_TuneCP5_GenSim+Digi+Reco+HARVEST+ALCA

**Summary**:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 36
 - DQMHistoTests: Total histograms compared: 2536530
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2536507
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 35 files compared)
 - Checked 151 log files, 37 edm output root files, 36 DQM output files

cvuosalo commented:
The logs from the new workflows look good. They seem to have run correctly.

silviodonato commented:
I've updated the PR description. @cms-sw/upgrade-l2 @cms-sw/pdmv-l2  do you have further comments?

chayanit commented:
+1

silviodonato commented:
merge
